Libraries/Opc.Ua.Security.Certificates/Extensions/X509SubjectAltNameExtension.cs

```diff
@@ -0,0 +1,446 @@
+ *
+ * OPC Foundation MIT License 1.00
+ * 
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ * 
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * The complete license agreement can be found here:
+ * http://opcfoundation.org/License/MIT/1.00/
+ * ======================================================================*/
+
+using System;
+using System.Collections.Generic;
+using System.Formats.Asn1;
+using System.Net;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+namespace Opc.Ua.Security.Certificates
+{
+    /// <summary>
+    /// The subject alternate name extension.
+    /// </summary>
+    /// <remarks>
+    /// 
+    /// id-ce-subjectAltName OBJECT IDENTIFIER::=  { id-ce 17 }
+    /// 
+    /// SubjectAltName::= GeneralNames
+    /// 
+    ///    GeneralNames::= SEQUENCE SIZE(1..MAX) OF GeneralName
+    /// 
+    ///    GeneralName ::= CHOICE {
+    ///        otherName                       [0] OtherName,
+    ///        rfc822Name[1]                   IA5String,
+    ///        dNSName[2]                      IA5String,
+    ///        x400Address[3]                  ORAddress,
+    ///        directoryName[4]                Name,
+    ///        ediPartyName[5]                 EDIPartyName,
+    ///        uniformResourceIdentifier[6]    IA5String,
+    ///        iPAddress[7]                    OCTET STRING,
+    ///        registeredID[8]                 OBJECT IDENTIFIER
+    ///        }
+    /// 
+    ///    OtherName::= SEQUENCE {
+    ///        type-id                         OBJECT IDENTIFIER,
+    ///        value[0] EXPLICIT ANY DEFINED BY type - id
+    ///        }
+    /// 
+    ///    EDIPartyName::= SEQUENCE {
+    ///        nameAssigner[0]                 DirectoryString OPTIONAL,
+    ///        partyName[1]                    DirectoryString
+    ///        }
+    /// 
+    /// </remarks>
+    public class X509SubjectAltNameExtension : X509Extension
```

Comment:
I made one comment about a ctor shape already, but there's a lot of how this type is defined that won't be true when we finally define it in .NET (not that I really know how it'll look yet, but I don't expect it will be IReadOnlyList properties for the subtypes, since it may be important the order things are defined in).

Reply:
TODO